### ISSUE TITLE
Build_hydro issues: missing buses

### ISSUE THREAD

**Issue by davide-f** [pz-max/pypsa-africa#706]
## Checklist

- [x] I am using the current [`main`](https://github.com/pypsa-meets-earth/pypsa-earth/tree/main) branch or the latest [release](https://github.com/pypsa-meets-earth/pypsa-earth/releases). Please indicate.
- [x] I am running on an up-to-date [`pypsa-earth` environment](https://github.com/pypsa-meets-earth/pypsa-earth/blob/main/envs/environment.yaml). Update via `conda env update -f envs/environment.yaml`.

## Describe the Bug

In some cases, hydro time series are not produced.
This needs to be addressed

## Error Message

*If applicable, paste any terminal output to help illustrating your problem.*
*In some cases it may also be useful to share your list of installed packages: `conda list`.*

```bash
Traceback (most recent call last):
  File "/data/davidef/git_world/pypsa-earth/.snakemake/scripts/tmp694caeih.add_electricity.py", line 779, in <module>
    attach_hydro(n, costs, ppl)
  File "/data/davidef/git_world/pypsa-earth/.snakemake/scripts/tmp694caeih.add_electricity.py", line 437, in attach_hydro
    assert missing_c.empty, (
AssertionError: 'resources/MU/renewable_profiles/profile_hydro.nc' is missing inflow time-series for at least one bus: 5, 6
```

**Comment by davide-f** [comment]
Similar problem appeared to @pz-max  also for Nigeria
See https://github.com/pypsa-meets-earth/pypsa-earth/pull/654#issuecomment-1537232523

**Comment by pz-max** [comment]
Can be reproduced by 'snakemake -j1 run_tests'

**Comment by davide-f** [comment]
By checking #709 , the feeling is that this issue is related to the parameter "clip_min_inflow" in build_renewable_profiles that is used to drop buses/values that are too small.
This may lead to some buses that are dropped.
I'll investigate it and it may be merged into #714 if so.
